Sources/PaymentsCore/api/EligibilityAPI.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+import PaymentsCore
+
+public class EligibilityAPI{
+    private var graphQLClient: GraphQLClient
+    private var coreConfig: CoreConfig
+    public init(coreConfig: CoreConfig){
+        self.coreConfig = coreConfig
+        graphQLClient = GraphQLClient(environment: coreConfig.environment)
+    }
+    
+    public func checkEligibility() async throws -> APIResult<Eligibility>{
```

Comment:
i think here we should do what we do in Android with `suspend` functions, which is the equivalent.
We return the value or throw an exception. @jaxdesmarais @sshropshire thoughts? Im not that well familiarized with iOS best practices.